### PR TITLE
Results_Engine: Update DisplayMeshResults to work with simplified interface requirements

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
check if cased result for error message

### NOTE: Depends on 

https://github.com/BHoM/BHoM/pull/1328
   
### Issues addressed by this PR

Changing so that error message is only mentioning result case if the result is a cased result.

- Tweaking the way the result values are extracted to increase the performance.

### Test files

Testing the MeshResultDisplay:

https://burohappold.sharepoint.com/:f:/s/BHoM/EmRmiGKVt5FFmQh7cOWCowQBR_MoUeNS5qPoemEjJmXrNw?e=sTvNF8

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

File: Results_Engine/Query/DisplayMeshResults.cs [hunk 1]
```diff
@@ -227,6 +266,32 @@ private static double ResultToValue(this IMeshElementResult r, string prop)
...
+
+            //Get the get method from the property
+            MethodInfo getMethod = propInfo?.GetGetMethod();
+
+            //Check that the property exists, has a get method and is of double type.
+            if (getMethod == null || propInfo.PropertyType != typeof(double))
+            {
+                //If incorrect type, raise error message with suggestions of property types that works for the current MeshElementResult type
+                Base.Compute.RecordError($"Property {prop} is not a valid property for results of type {typeof(T).Name}." + 
+                    $"Try one of the following: {typeof(T).GetProperties(BindingFlags.DeclaredOnly | BindingFlags.Public | BindingFlags.Instance).Where(x => x.PropertyType == typeof(double)).Select(x => x.Name).Aggregate((a, b) => a + " ," + b)}");
```
Comment (alelom) [inline Results_Engine/Query/DisplayMeshResults.cs:283]:
```suggestion
                    $"Try one of the following: {typeof(T).GetProperties(BindingFlags.DeclaredOnly | BindingFlags.Public | BindingFlags.Instance).Where(x => x.PropertyType == typeof(double)).Select(x => x.Name).Aggregate((a, b) => a + ", " + b)}");
```

**Review by alelom — APPROVED** [review]
Works well and decouples effectively! LGTM!

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Comment by IsakNaslundBh** [comment]
@BHoMBot check dataset-compliance
@BHoMBot check copyright-compliance
@BHoMBot check branch-compliance

**Pull request merged**